src/screens/settings/components/SettingsContainer.js

```diff
@@ -0,0 +1,230 @@
+import React, { useState, useEffect, useCallback, useContext } from 'react';
+
+// Components
+import { StatusBar, Alert, StyleSheet, View } from 'react-native';
+import {
+  Container,
+  Header,
+  Title,
+  Content,
+  Footer,
+  FooterTab,
+  Button,
+  Left,
+  Right,
+  Body,
+  Icon,
+  Text,
+  Form,
+  Picker,
+  H1,
+  H3,
+} from 'native-base';
+import NameBar from './NameBar';
+
+// Utils
+import {
+  getUser,
+  setUser,
+  clearWordList,
+  getUserWordList,
+  getAllUsers,
+} from '../../../utils/storage';
+
+// Context
+import UserContext from '../../../contexts/UserContext';
+
+// Styles
+const styles = StyleSheet.create({
+  wordListContainer: {
+    display: 'flex',
+    flexDirection: 'row',
+    padding: 5,
+  },
+  wordList: {
+    flex: 5,
+    flexDirection: 'row',
+  },
+  refresh: {
+    flex: 1,
+    flexDirection: 'row',
+  },
+  wordLevel: {
+    padding: 15,
+  },
+});
+
+// Vars
+const NEW_USER = 'New User';
+
+const updateData = (inputText, setName) => {
+  setName(inputText);
+  setUser(inputText);
+};
+
+const getData = async () => getUser();
+
+const clearWords = async () => {
+  clearWordList();
+};
+
+const handleClearButtonPress = () => {
+  Alert.alert(
+    'Are you sure you want to delete all of your words?',
+    'This action is irreversible. You will lose all of your word data.',
+    [
+      {
+        text: 'Cancel',
+        onPress: () => {},
+        style: 'cancel',
+      },
+      { text: 'Delete All Words', onPress: clearWords, style: 'danger' },
+    ],
+  );
+};
+
+const getUsers = async () => {
+  const users = await getAllUsers();
+  return users;
+};
+
+const SettingsContainer = () => {
+  const [name, setName] = useContext(UserContext);
```

Comment:
Kinda curious about the `UserContext` here, but I could be missing something — it seems like on both these views, you fetch the user and set it in the context, and on all future views you'd need to do the same. Is it possible to

Reply:
accidentally hit "comment" while still thinking about this. Gonna delete 😆